bench/bench.hs

```diff
@@ -106,9 +106,10 @@ bench_atoi
  where
   go_all nm f
    = bgroup nm
-   $ fmap (\str -> bench (show str) $ nf f str)
+   $ fmap (\str -> bench (show str) $ nf f (pad str))
    [ "0", "1", "-123", "123456", "-12345678", "1234567890", "-123456789012345", "1234567890123456789", "12345678901234567890"
    ]
+  pad str = str <> " (padding added) "
```

Comment:
I added this padding because hopefully in most real uses your buffer will be large enough to load the whole 128bits at a time. however it didn't make much difference to the benchmark results